File: apps/backend/services/Identity/Identity.Domain/Enums/TenantType.cs
namespace Identity.Domain.Enums;

/// <summary>
/// Classifies what kind of business a tenant runs. Used only at tenant-creation time
/// to seed sensible <see cref="Entities.FeatureFlags"/> defaults (see
/// <see cref="Entities.FeatureFlags.CreateDefaults"/>) — after creation, flags can be
/// toggled independently of this value, so this is not read anywhere else.
/// </summary>
public enum TenantType
{
    Rental = 1,
    Farming = 2
}

File: apps/mobile/lib/config/tenant/tenant_type.dart
/// Mirrors the backend's Identity.Domain.Enums.TenantType — classifies what kind
/// of business the tenant runs. Delivered at build time via TENANT_TYPE in the
/// tenant's .env JSON file (same mechanism as the other tenant/feature config),
/// fetched from the backend by set_tenant_branding.sh before the build runs.
///
/// Only used to decide which pages/tabs the app was built with — it does not
/// gate anything at runtime. FeatureFlags (see feature_flags.dart) still control
/// which of those tabs are actually shown, since flags can be toggled per-tenant
/// independently of this type.
enum TenantType { rental, farming }

TenantType tenantTypeFromString(String raw) => switch (raw) {
      'Farming' => TenantType.farming,
      _ => TenantType.rental,
    };
